### ISSUE TITLE
Due dates for divisional stages

### ISSUE THREAD

**Issue by lmullane** [bcgov/foi-flow#1674]
* As a Ministry Coordinator
* I want to be able to track due dates at the divisional stage
* so that I know when I need to follow up with divisions

**Assumptions & Scope**	
What are the assumptions for this story?

Only need due dates for 3 divisional stages: Gathering Records, Awaiting Harms, Pending Sign Off.

Each division should have a separate due date for the 3 stages.

Each due date stage should include an optional field for an eApproval number.

Due date, and e-approval number will NOT be mandatory for save.

What is IN scope?
Add a due date to divisional stages: Gathering Records, Awaiting Harms, Pending Sign Off.

Field for eApprovals for the 3 divisional stages.

What is NOT in scope?
Notifications for due dates for divisional stages. 

**Acceptance Criteria**	

**Scenario 1: Due Date and E-Approval Fields**
* GIVEN a Ministry coordinator is on the review request page
* AND they have selected a division
* WHEN they select a divisional stage that is either: gathering records, pending harms, or awaiting sign-off
* THEN two additional fields will appear: a free text field for E-Apps/Other 
* AND a field for a due date, that includes a calendar picker consistent with other calendar pickers used in the application
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/768088423

**Scenario 2: Due Date Calendar Picker - Activation**
* GIVEN a Ministry coordinator has selected a divisional stage that reveals the due date calendar picker
* WHEN they click on the calendar icon
* THEN the will active the calendar picker, consistent with other pickers used across the application

**Scenario 3: Select Due Date**
* GIVEN a Ministry coordinator has activated the calendar picker on a divisional stage
* WHEN they select a date
* THEN the date will appear in the YYYY MM DD Format

**Scenario 4: Due Date Selected Cannot be in Past**
* GIVEN a Ministry coordinator has activated the calendar picker on a divisional stage
* WHEN they attempt to select a date in the past
* THEN they will not be able to select that date

**Scenario 5: E-Apps/Other Ref Number**
* GIVEN a Ministry coordinator has selected a divisional stage that reveals the due date calendar picker
* WHEN they click on the space for E-Apps/Other
* THEN they will be able to input text into the box, up to a maximum of 12 characters

**Scenario 6: Save Changes**
* GIVEN a Ministry coordinator has selected a divisional stage that reveals the due date calendar picker
* AND they have filled in the date and/or the e-apps/other field
* AND all other mandatory fields have been captured in the request
* WHEN they click save changes
* THEN the save toast will appear in the top right corner
* AND The divisional tracking changes will be saved

**Scenario 7: Divisional Comment Update with Due Date and E-App**
* GIVEN a Ministry coordinator has saved a divisional stage that includes a due date and/or an e-apps field filled in
* WHEN the automated comment is added to the comment log
* THEN the message will be: "[Division Name] division has been updated to [Stage] E-App/Other reference Number: XXXXXX Due on YYYY MM DD"

**Scenario 8: Divisional Comment Update with no Due Date or E-App Reference**
* GIVEN a Ministry coordinator has saved a divisional stage that does not includes a due date or an e-apps field filled in
* WHEN the automated comment is added to the comment log
* THEN the message will be: "[Division Name] division has been updated to [Stage]

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/768088423

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by lmullane** [comment]
@JHarrietha-AOT, I've updated the assumptions to include the field for eApprovals.

As discussed, I will draft the AC once you have created the designs.

cc @arielleandrews, @m-prodan, @mpilchar

**Comment by JHarrietha-AOT** [comment]
@m-prodan Updated Designs: https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/768088423

cc: @abin-aot @divyav-aot. Matt will be writing the AC today for these, but happy to hop on a call to walk you through them in you have questions and want to be able to estimate

**Comment by JHarrietha-AOT** [comment]
Comment Updates (If Due date/E-App or Other is input)
"_Education Programs_ division has been updated to stage _Gathering Records_ E-App/Other reference Number _0000000_ Due on _YYYY MM DD_"

**Comment by m-prodan** [comment]
Story update - I included a character limit of 12 for e-apps, to cover all potential use cases. I also added two scenarios for automated comment updates ,which may need to be separated out as a separate user story. I will tag team for refinement tomorrow!

**Comment by mpilchar** [comment]
Needs to be completed before Apr 8 pen test

**Comment by lmullane** [comment]
Great work, @aparna-aot and @divyav-aot .

All AC passed as expected.
